docs/frontend-system/building-plugins/migrating.md

```diff
@@ -0,0 +1,189 @@
+---
+id: migrating
+title: Migrating Plugins
+sidebar_label: Migration Guide
+# prettier-ignore
+description: How to migrate an existing frontend plugin to the new frontend system
+---
+
+This guide allows you to migrate a frontend plugin and its own components, routes, apis to the new frontend system.
+
+The main concept is that routes, components, apis are now extensions. You can use the appropriate extension creators to migrate all of them to extensions.
+
+## Pages
+
+Pages that were previously created using the `createRoutableExtension` extension function can be migrated to the new Frontend System using the `createPageExtension` extension creator, exported by `@backstage/frontend-plugin-api`.
+
+For example, given the following page:
+
+```ts
+export const FooPage = fooPlugin.provide(
+  createRoutableExtension({
+    name: 'FooPage',
+    component: () => import('./components').then(m => m.FooPage),
+    mountPoint: rootRouteRef,
+  }),
+);
+```
+
+it can be migrated as the following:
+
+```tsx
+import { createPageExtension } from '@backstage/frontend-plugin-api';
+import {
+  compatWrapper,
+  convertLegacyRouteRef,
+} from '@backstage/core-compat-api';
+
+const fooPage = createPageExtension({
+  defaultPath: '/foo',
+  // you can reuse the existing routeRef
+  // by wrapping into the convertLegacyRouteRef.
+  routeRef: convertLegacyRouteRef(rootRouteRef),
+  // these inputs usually match the props required by the component.
+  loader: ({ inputs }) =>
+    import('./components/').then(m =>
+      // The compatWrapper utility allows you to use the existing
+      // legacy frontend utilities used internally by the components.
+      compatWrapper(<m.FooPage />),
+    ),
+});
+```
+
+## Components
+
+The equivalent utility to replace components created with `createComponentExtension` is `createExtension` from `@backstage/frontend-plugin-api`. However, we recommend searching for an appropriate extension first, considering using `createExtension` as the last option for more complex scenarios.
```

Comment:
It's even the case that if there is no existing extension creator for the new system then there's no way to migrate that extension yet. You won't be able to use `createExtension` since there won't be any input that accepts your extension. If such an input exists then it should always be accompanied by an extension creator.